rfcs/min-relations.md

```diff
@@ -0,0 +1,438 @@
+Here's an example of how you might do so:
+
+```rust
+fn love_potion(
+    commands: mut Commands,
+    query: Query<Entity, (With<Relation<&Loves>>, With<Cute>)>,
+    player_query: Query<Entity, With<Player>>){
+    
+    let player = player.single().unwrap();
+
+    for victim in query.iter(){
+        // change_target() and change_source() preserve the relation's data
+        // Only one relation of a given kind can exist between a source and a target;
+        // these command applies to any relations that match,
+        // and overwrite any conflicting relations
+        commands.entity(victim).change_target::<Loves>(player); // This is unethical!!
+    }
+}
+
+fn adoption(
+    commands: mut Commands,
+    // As with components, you can query for relations that may or may not exist
+    query: Query<(Entity, &NewOwner, Option<Relation<Owns>>), With<Kitten>>,
+) {
+   for (kitten, new_owner, previous_ownership) in query.iter(){
+       // Changing the target or the source will fail silently if no appropriate relation exists
+       match previous_ownership {
+           // The first parameter is the data, the second is the old target
+           Some(_,_) => commands.entity(kitten).change_source::<Owns>(new_owner); // uwu :3
+           None => commands.entity(kitten).insert_relation(Owns::default(), new_owner); // uwu!!
+       }
+   } 
+}
+```
+
+### Advanced relation filters
+
+As the savvy reader may have guessed, we can combine relation filters in arbitrarily complex ways.
+While most of you may never need to plumb the dizzying heights of compound relation filtering,
+the DBAs in the audience will surely both thank us for our efforts and deride our naive implementations.
+
+Fortunately, there are several simple and convenient API extensions that you might want to use.
+The first of these is the ability to filter by the source entity as well:
+
+```rust
+fn 
+
+```
+
+Next, you may wish to operate over entire groups of entities at once:
+
+```rust
+fn 
+
+```
+
+From time-to-time, we may care about *excluding* entities who have relations to certain targets:
+
+```rust
+fn blackball(){
+
+}
+
+```
+
+`any_of` and `all_of` can be used to collect groups of entities into a single filter.
+`any_of` uses **or semantics**, returning any entity if any of the filters are met,
+while `all_of` uses **and semantics**, rejecting entities who do not meet every specified filter.
+
+```rust
+fn 
+
+```
+
+We can filter on multiple types of relations at once by chaining together our `.filter_relation` methods before we call `.build()`.
+
+```rust
+fn 
+
+```
+
+Chaining filters in this way uses "and" semantics, just like `all_of`.
+
+While relations filters are reset each time you leave a system, you may want to reset them sooner,
+commonly when performing multiple subqueries in bevy. To do so, simply call `query.reset_filter::<R>()` or `query.reset_filters()`.
```

Comment:
Why do filters need a reset? What kind of state do they hold? I feel like starting fresh iterator on the query should always give me the same thing.

I presume the idea here is to "remove previously added relation filters". In that case, how about `query.remove_relation_filter::<R>()` and `query.clear_relation_filters()`?

Also this remove method accepts a relation type and nothing else. Does that mean I can only add a single filter per given relation type in the first place? Or does it remove all added filters with this relation type?